### ISSUE TITLE
Psr7\readline has trouble with non-native EOLs

### ISSUE THREAD

**Issue by avltree** [guzzle/psr7#77]
The condition to check if we got to the end of the line in `readline` is defined as:

```
if ($byte == PHP_EOL || ++$size == $maxLength - 1) {
// ...
```

It will work properly if the stream was generated by a server, which uses the same type of line endings as the system on which the stream is being processed.

For example: an application running on a Linux server returns a stream using `\n` as EOL. `readline` is used to read this stream on a Windows server, where `PHP_EOL === "\r\n"`. Because `$byte == "\r\n"` will never be true, `readline` will read the whole stream (or to `$maxLength`, if provided) instead of only the first line.

**Comment by mtdowling** [comment]
What do you suggest we do? Add an optional line ending argument?

**Comment by avltree** [comment]
Yes, that's probably a good idea (if it doesn't break any standards, fo course).

**Comment by mtdowling** [comment]
It would not violate the PSR7 standards. I'd be good with that change if you or someone else wants to tackle it.

**Comment by Tobion** [comment]
I already fixed this in #81

**Comment by mtdowling** [comment]
Thanks, @Tobion
